File: Tutoriels/Python/Projets/src/manager/GPrint.py
#================================================
class GPrint:
    #================================================
    m_instance = None
    #================================================
    def __init__(self):
        if GPrint.m_instance == None:
            GPrint.m_instance = self        
    #================================================
    @staticmethod
    def Instance():
        if GPrint.m_instance == None:
            GPrint()
        return GPrint.m_instance
    #================================================
    def printData(self, data, name=""):
        if(name): print(name + ":"),
        print(data) 
#================================================

File: Tutoriels/CSharp/Projets/src/manager/GPrint.cs
//===============================================
using System;
//===============================================
public class GPrint {
    //===============================================
    private static GPrint m_instance = null;
    //===============================================
    private GPrint() {

    }
    //===============================================
    public static GPrint Instance() {
        if(m_instance == null) {
            m_instance = new GPrint();
        }
        return m_instance;
    }
    //===============================================
    public void Print(string data) {
        Console.Write(data + "\n");
    }
    //===============================================
}
//===============================================
